proto/kuksa/val/v2/val.proto

```diff
@@ -0,0 +1,227 @@
+  rpc Subscribe(SubscribeRequest) returns (stream SubscribeResponse);
+
+  // Actuate a single actuator
+  //
+  // Returns (GRPC error code):
+  //   NOT_FOUND if the actuator does not exist.
+  //   PERMISSION_DENIED if access is denied for of the actuator.
+  //   UNAVAILABLE if there is no provider currently providing the actuator
+  //   INVALID_ARGUMENT
+  //       - if the data type used in the request does not match
+  //            the data type of the addressed signal
+  //       - if the requested value is not accepted,
+  //            e.g. if sending an unsupported enum value
+  rpc Actuate(ActuateRequest) returns (ActuateResponse);
+
+  // Actuate simultaneously multiple actuators.
+  // If any error occurs, the entire operation will be aborted
+  // and no single actuator value will be forwarded to the provider.
+  //
+  // Returns (GRPC error code):
+  //   NOT_FOUND if any of the actuators are non-existant.
+  //   PERMISSION_DENIED if access is denied for any of the actuators.
+  //   UNAVAILABLE if there is no provider currently providing an actuator
+  //   INVALID_ARGUMENT
+  //       - if the data type used in the request does not match
+  //            the data type of the addressed signal
+  //       - if the requested value is not accepted,
+  //            e.g. if sending an unsupported enum value
+  rpc BatchActuate(BatchActuateRequest) returns (BatchActuateResponse);
+
+  // List metadata of signals matching the request.
+  //   NOT_FOUND if the specified root branch does not exist.
+  rpc ListMetadata(ListMetadataRequest) returns (ListMetadataResponse);
+
+  // Publish a signal value. Used for low frequency signals (e.g. attributes).
+  //
+  // Returns (GRPC error code):
+  //   NOT_FOUND if any of the signals are non-existant.
+  //   PERMISSION_DENIED
+  //       - if access is denied for any of the signals.
+  //       - if the signal is already provided by another provider.
+  //   INVALID_ARGUMENT
+  //       - if the data type used in the request does not match
+  //            the data type of the addressed signal
+  //       - if the published value is not accepted,
+  //            e.g. if sending an unsupported enum value
+  rpc PublishValue(PublishValueRequest) returns (PublishValueResponse);
+
+  // Open a stream used to provide actuation and/or publishing values using
+  // a streaming interface. Used to provide actuators and to enable high frequency
+  // updates of values.
+  //
+  // The open stream is used for request / response type communication between the
+  // provider and server (where the initiator of a request can vary).
+  // Errors are communicated as messages in the stream.
+  rpc OpenProviderStream(stream OpenProviderStreamRequest) returns (stream OpenProviderStreamResponse);
+
+  // Get server information
+  rpc GetServerInfo(GetServerInfoRequest) returns (GetServerInfoResponse);
+}
+
+message GetValueRequest {
+  SignalID signal_id = 1;
```

Comment:
Maybe it would be better having two separate requests: `GetValue(ByName)Request` and `GetValueByIdRequest`.

I mean, we introduced the single signal get for making life easier for a developer, but know we put "the burden" on them to make this extra set step for the oneof ...